### PR TITLE
Dont save uploaded testcases to regression corpus

### PR DESCRIPTION
**Pull request by jonathanmetzman**

### CODE REVIEW

File: src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py [hunk 1]
```diff
@@ -405,6 +405,15 @@ def _set_regression_testcase_upload_url(
   if not fuzz_target:
     # No work to do, only applicable for engine fuzzers.
     return
+
+  if testcase.uploader_email:
+    logs.log_warn('Not saving uploaded to regression corpus.')
```
Comment (alhijazi) [inline src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py:410]: uploaded testcase?

File: src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py [hunk 2]
```diff
@@ -405,6 +405,15 @@ def _set_regression_testcase_upload_url(
...
+    return
+  upload_metadata = data_types.TestcaseUploadMetadata.query(
+      data_types.TestcaseUploadMetadata.testcase_id == testcase.key.id()).get()
+  if upload_metadata:
```
Comment (alhijazi) [inline src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py:414]: Could you please add a comment explaining what the presence of `upload_metadata` means and why it leads to skippping the upload?

File: src/clusterfuzz/_internal/datastore/data_types.py [hunk 3]
```diff
@@ -576,6 +576,11 @@ class Testcase(Model):
   # Note that the number is specific to the repository.
   github_issue_num = ndb.IntegerProperty()
 
+  # Whether the testcase was uploaded.
+  # True by default since this will determine what is put in the regression
+  # corpus.
+  uploaded = ndb.BooleanProperty(default=True)
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/datastore/data_types.py:582]: Can we rename this to "untrusted"? 

I don't think we're inherently interested in whether this was uploaded by somebody or not. Rather, it's the fact that it's untrusted.

Comment (oliverchang) [inline src/clusterfuzz/_internal/datastore/data_types.py:582]: IS the rename not pushed for this file?

File: src/clusterfuzz/_internal/tests/core/datastore/data_types_test.py [hunk 4]
```diff
@@ -61,6 +61,8 @@ def test_put(self):
         'es', 'es.1', 's', 's.1', 'b', 'b.3', 'stable', 'beta',
         'extended_stable'
     }, set(testcase.impact_version_indices))
+    # Assert that this is true by default.
+    self.assertTrue(testcase.uploaded)
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/tests/core/datastore/data_types_test.py:65]: and also here.

File: src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py [hunk 5]
```diff
@@ -405,6 +405,22 @@ def _set_regression_testcase_upload_url(
   if not fuzz_target:
     # No work to do, only applicable for engine fuzzers.
     return
+
+  if not testcase.trusted:
+    logs.log_warn('Not saving untrusted testcase to regression corpus.')
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py:410]: should we return here?

**Review by oliverchang — APPROVED** [review]

**Pull request merged**